aviary/subsystems/propulsion/engine_deck.py

```diff
@@ -1456,15 +1457,19 @@ def build_mission(self, num_nodes, aviary_inputs):
                                    promotes_inputs=['*'],
                                    promotes_outputs=['*'])
 
-        if self.prop_model:
+        if self.prop_model is True:
+            self._add_HS_prop(engine_group, num_nodes)
+        elif self.prop_model is None:
+            self._add_dummy_prop(engine_group, num_nodes)
+        elif isinstance(self.prop_model, (om.Group, Component)):
```

Comment:
I suppose you could just check if it is a `System` so that it does just 1 check.

Reply:
Good call, changed that to `System`